hilti/toolchain/src/compiler/jit.cc

```diff
@@ -176,15 +201,29 @@ hilti::Result<Nothing> JIT::_compile() {
 
     auto cc_files = _files;
 
+    // Remember generated files and remove them on all exit paths.
+    std::shared_ptr<std::vector<hilti::rt::filesystem::path>>
+        cc_files_generated(new std::vector<hilti::rt::filesystem::path>(), [](const auto* cc_files_generated) {
+            for ( const auto& cc : *cc_files_generated ) {
+                std::error_code ec;
+                HILTI_DEBUG(logging::debug::Jit, util::fmt("removing temporary file %s", cc));
+                hilti::rt::filesystem::remove(cc, ec);
+
+                if ( ec )
+                    HILTI_DEBUG(logging::debug::Jit,
+                                util::fmt("could not remove temporary file %s: %s", cc, ec.message()));
+            }
+
+            delete cc_files_generated;
+        });
+
```

Comment:
I find this quite unreadable ... I think scope-guard-style object would improve this.

Reply:
I'll add a helper class.